src/WebJobs.Script.WebHost/Security/KeyManagement/SecretManager.cs

```diff
@@ -39,6 +39,12 @@ public SecretManager(ISecretsRepository repository, IKeyValueConverterFactory ke
             _repository.SecretsChanged += OnSecretsChanged;
             _logger = logger;
 
+            //Hacky way to add loger to migration repository.
+            if (_repository is BlobStorageSecretsMigrationRepository)
```

Comment:
We should be able to cleanly pass that at construction time by updating the factory instead, right?

Reply:
good idea